api/src/controllers/feature-flag.controller.ts

```diff
@@ -0,0 +1,115 @@
+import { FeatureFlagUpdate } from '../dtos/feature-flags/feature-flag-update.dto';
+import { defaultValidationPipeOptions } from '../utilities/default-validation-pipe-options';
+import { IdDTO } from '../dtos/shared/id.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { PermissionTypeDecorator } from '../decorators/permission-type.decorator';
+import { OptionalAuthGuard } from '../guards/optional.guard';
+import { PermissionGuard } from '../guards/permission.guard';
+import { ApiKeyGuard } from '../guards/api-key.guard';
+
+@Controller('featureFlags')
+@ApiTags('featureFlags')
+@UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+@ApiExtraModels(
+  FeatureFlagAssociate,
+  FeatureFlagCreate,
+  FeatureFlagUpdate,
+  IdDTO,
+)
+@PermissionTypeDecorator('featureFlags')
+@UseGuards(OptionalAuthGuard, PermissionGuard)
+export class FeatureFlagController {
+  constructor(private readonly featureFlagService: FeatureFlagService) {}
+
+  @Get()
+  @ApiOperation({ summary: 'List of feature flags', operationId: 'list' })
+  @ApiOkResponse({ type: FeatureFlag, isArray: true })
+  async list(): Promise<FeatureFlag[]> {
+    return await this.featureFlagService.list();
+  }
+
+  @Post()
+  @ApiOperation({
+    summary: 'Create a feature flag',
+    operationId: 'create',
+  })
+  @ApiOkResponse({ type: FeatureFlag })
+  @UseGuards(ApiKeyGuard)
+  async create(@Body() featureFlag: FeatureFlagCreate): Promise<FeatureFlag> {
+    return await this.featureFlagService.create(featureFlag);
+  }
+
+  @Put()
+  @ApiOperation({
+    summary: 'Update a feature flag',
+    operationId: 'update',
+  })
+  @ApiOkResponse({ type: FeatureFlag })
+  @UseGuards(ApiKeyGuard)
+  async update(@Body() featureFlag: FeatureFlagUpdate): Promise<FeatureFlag> {
+    return await this.featureFlagService.update(featureFlag);
+  }
+
+  @Delete()
+  @ApiOperation({
+    summary: 'Delete a feature flag by id',
+    operationId: 'delete',
+  })
+  @ApiOkResponse({ type: SuccessDTO })
+  @UseGuards(ApiKeyGuard)
+  async delete(@Body() dto: IdDTO): Promise<SuccessDTO> {
+    return await this.featureFlagService.delete(dto.id);
+  }
+
+  @Put(`associateJurisdictions`)
+  @ApiOperation({
+    summary: 'Associate and disassociate jurisdictions with a feature flag',
+    operationId: 'associateJurisdictions',
+  })
+  @ApiOkResponse({ type: FeatureFlag })
+  @UseGuards(ApiKeyGuard)
+  async associateJurisdictions(
+    @Body() featureFlagAssociate: FeatureFlagAssociate,
+  ): Promise<FeatureFlag> {
+    return await this.featureFlagService.associateJurisdictions(
+      featureFlagAssociate,
+    );
+  }
+
+  @Get(`:featureFlagId`)
+  @ApiOperation({
+    summary: 'Get a feature flag by id',
+    operationId: 'retrieve',
+  })
+  @ApiOkResponse({ type: FeatureFlag })
+  @UseGuards(ApiKeyGuard)
+  async retrieve(
+    @Param('featureFlagId', new ParseUUIDPipe({ version: '4' }))
+    featureFlagId: string,
+  ): Promise<FeatureFlag | null> {
```

Comment:
do you need the `| null`

Reply:
Huh guess not. I followed the jurisdiction controller when making most of this, but looking at other controllers it doesn't seem we do this. Will remove